rspace/src/main/scala/coop/rchain/rspace/channelStore/instances/ChannelStoreImpl.scala

```diff
@@ -0,0 +1,66 @@
+package coop.rchain.rspace.channelStore.instances
+
+import cats.effect.Sync
+import cats.implicits._
+import coop.rchain.rspace.Hasher.{hashContinuationsChannels, hashDataChannel, hashJoinsChannel}
+import coop.rchain.rspace.channelStore.{ChannelHash, ChannelStore, ContinuationHash, DataJoinHash}
+import coop.rchain.rspace.history.Store
+import coop.rchain.rspace.internal.{toOrderedByteVectors, RichAttempt}
+import coop.rchain.rspace.{Blake2b256Hash, StableHashProvider}
+import coop.rchain.shared.Serialize
+import scodec.Codec
+import scodec.codecs.{discriminated, uint2}
+
+object ChannelStoreImpl {
+  def apply[F[_]: Sync, C](store: Store[F], sc: Serialize[C], codecC: Codec[C]) =
+    ChannelStoreImpl(store, sc, codecC)
+
+  final case class ChannelStoreImpl[F[_]: Sync, C](
+      store: Store[F],
```

Comment:
Using KeyValueStore should be much nicer because you don't have to think about closing and we want to remove Store eventually.
Even better with `KeyValueTypedStore[F, Blake2b256Hash, ChannelHash]` it's almost just like using a Map.
For now you need to supply codecs as it is but later we can add some "communication" to underlying codes in typed store.

Reply:
Another pr.